udemy_exercises/bank/Bank.java

```diff
@@ -1,10 +1,114 @@
+        Branch branch = findBranch(branchName);
+        if (branch != null) {
+            return branch.addCustomerTransaction(customerName, amount);
+        }
+        return false;
+    }
+
+    private Branch findBranch(String branchName) {
+        for (int i = 0; i < branches.size(); i++) {
+            Branch checkedBranch = this.branches.get(i);
+            if (checkedBranch.getName().equals(branchName)) {
+                return checkedBranch;
+            }
+        }
+        return null;
+    }
+
+    public boolean listCustomers(String branchName, boolean showTransactions) {
+        Branch branch = findBranch(branchName);
+        if (branch != null) {
+            System.out.println("Customer details for branch" + branch.getName() + ":");
+            ArrayList<Customer> branchCustomers = branch.getCustomers();
+            for (int i = 0; i < branchCustomers.size(); i++) {
+                Customer branchCustomer = branchCustomers.get(i);
+                System.out.println("Customer: " + branchCustomer.getName() + "[" + (i + 1) + "]");
+                if (showTransactions) {
+                    System.out.println("Transactions: ");
+                    ArrayList<Double> transactions = branchCustomer.getTransactions();
+                    for (int j = 0; j < transactions.size(); j++) {
+                        System.out.println("[" + (j + 1) + "] Amount: " + transactions.get(j));
+                    }
+                }
+            }
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static void main(String[] args) {
+        Bank bank = new Bank("National Bank");
+
+        if (bank.addBranch("Dallas")) {
+            System.out.println("Dallas branch created.");
+        }
+
+        bank.addCustomer("Dallas", "Sarah", 75.25);
+        bank.addCustomer("Dallas", "Rachael", 1350.89);
+        bank.addCustomer("Dallas", "Tucker", 4.13);
+
+        bank.addBranch("Vail");
+        bank.addCustomer("Vail", "Kyle", 5601.92);
+
+        bank.addCustomerTransaction("Dallas", "Sarah", 500.25);
+        bank.addCustomerTransaction("Dallas", "Sarah", 119.24);
+        bank.addCustomerTransaction("Dallas", "Rachael", -50.45);
+
+        bank.listCustomers("Dallas", true);
+        bank.listCustomers("Vail", true);
+
+        bank.addBranch("Austin");
+
+        if (!bank.addCustomer("Austin", "Ryan", 10.67)) {
+            System.out.println("Error. Austin branch does not exist.");
+        }
+        if (!bank.addBranch("Dallas")) {
+            System.out.println("Dallas branch already exists.");
+        }
+        if (!bank.addCustomerTransaction("Dallas", "Amanda", 46.78)) {
+            System.out.println("Customer does not exist.");
+        }
+        if (!bank.addCustomer("Dallas", "Sarah", 31.67)) {
+            System.out.println("Customer Sarah already exists.");
+        }
+    }
```

Comment:
Lots of If/elses here. I know you're much better at designing a Java app now. Not that this doesn't work but the design is clunky